presto-main/src/test/java/com/facebook/presto/execution/buffer/TestPageCodecMarker.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.buffer;
+
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestPageCodecMarker
```

Comment:
I find these tests hard to read. The complexity of the test code seems to exceed the complexity of the enum being tested. I think testing the only two possible values (as suggested in an earlier comment) and their combinations is appropriate and ensures that production use cases are covered.

Reply:
I'll add a more specific concrete test method, but feel pretty strongly about the exhaustive checking approach as a long-term safeguard against regressions that may be introduced with new `PageCodecMarker` values. For instance, adding a new marker type but accidentally reusing the bit position from another value or extending the masking logic to include the compression codec alongside the `COMPRESSED` marker might easily introduce conflicting marker sets that wouldn't be explicitly combined in concrete tests.

I agree that it makes the test seem more complex than the `PageCodecMarker` implementation itself, but I've caught enough logical errors using a more "property based testing" style approach that I'm convinced they're worth at least a small complexity cost.

As always, if I haven't convinced you then I'm fine removing it to get the PR merged. I think I've addressed all your other comments.